core/sql/common/DateTimeType.cpp

```diff
@@ -171,6 +172,8 @@ NABoolean DatetimeType::isEncodingNeeded() const
 
 NABoolean DatetimeType::operator==(const NAType& other) const
 {
+  DatetimeType &otherDT = (DatetimeType&)other;
```

Comment:
Not sure why this variable is defined. We don't use it anywhere. Doesn't hurt anything though.